filter/hash/filterhash.go

```diff
@@ -0,0 +1,196 @@
+)
+
+// hashAlgo defines a hash implementing hash.Hash
+type hashAlgo struct {
+	Name string           // name of algo
+	Init func() hash.Hash // init func for hash method
+}
+
+// hash32Algo defines a hash implementing hash.Hash32
+type hash32Algo struct {
+	Name string             // name of algo
+	Init func() hash.Hash32 // init func for hash method
+}
+
+// hash32Algos is a list of supported hash.Hash32 algorithms
+var hash32Algos = []hash32Algo{
+	{"adler32", adler32.New},
+	{"fnv32a", fnv.New32a},
+}
+
+// hashAlgos is a list of supported hash.Hash algorithms
+var hashAlgos = []hashAlgo{
+	{"sha1", sha1.New},
+	{"sha256", sha256.New},
+	{"md5", md5.New},
+	{"fnv128a", fnv.New128a},
+}
+
+// InitHandler initialize the filter plugin
+func InitHandler(ctx context.Context, raw *config.ConfigRaw) (config.TypeFilterConfig, error) {
+	conf := DefaultFilterConfig()
+	if err := config.ReflectConfig(raw, &conf); err != nil {
+		return nil, err
+	}
+
+	if len(conf.Source) == 0 {
+		return nil, errors.New("hash: no source fields")
+	}
+	if len(conf.Target) == 0 {
+		return nil, errors.New("hash: no destination field")
+	}
+
+	err := initHashConfig(&conf)
+
+	if err == nil {
+		goglog.Logger.Debug("hash filter initialized")
+	}
+	return &conf, err
+}
+
+// initHashConfig initializes hash configuration. Moved to its own function so it can be tested.
+func initHashConfig(conf *FilterConfig) error {
+	var err error
+
+	// we either prepare a hash or hash32 algo
+	for _, v := range hashAlgos {
+		if v.Name == conf.Kind {
+			conf.hash = v.Init()
+			break
+		}
+	}
+	if conf.hash == nil {
+		for _, v := range hash32Algos {
+			if v.Name == conf.Kind {
+				conf.hash32 = v.Init
+				break
+			}
+		}
+	}
+	if conf.hash == nil && conf.hash32 == nil {
+		msg := fmt.Sprint("filterhash unsupported hash algo: ", conf.Kind)
+		err = errors.New(msg)
+		return err
+	}
+
+	switch conf.Format {
+	case "binary":
+		conf.outputFormat = outputBinary
+	case "base64":
+		conf.outputFormat = outputBase64
+	case "int":
+		conf.outputFormat = outputBigInt
+	default:
+		conf.outputFormat = outputHex
+	}
+	return nil
+}
+
+// Event is the main filter event
+func (f *FilterConfig) Event(ctx context.Context, event logevent.LogEvent) (logevent.LogEvent, bool) {
+	// concat all fields that should be hashed together, make hash and set value
+	source := f.getUnhashedString(&event)
+	hash := f.makeHash(source)
+	event.SetValue(f.Target, hash)
+	return event, true
+}
+
+// getUnhashedString returns all configured source fields concatenated together
+func (f *FilterConfig) getUnhashedString(event *logevent.LogEvent) string {
+	var sourceString string
+	for _, v := range f.Source {
+		field := event.Get(v)
+		sourceString = fmt.Sprintf("%v%v", sourceString, field)
+	}
+	return sourceString
+}
+
+// i32tob convert uint32 to byte array
+func i32tob(val uint32) []byte {
+	r := make([]byte, 4)
+	for i := uint32(0); i < 4; i++ {
+		r[i] = byte((val >> (8 * i)) & 0xff)
+	}
+	return r
+}
```

Comment:
https://stackoverflow.com/questions/16888357/convert-an-integer-to-a-byte-array
Better?